app/src/main/java/io/appium/settings/Settings.java

```diff
@@ -70,12 +99,152 @@ public void onCreate(Bundle savedInstanceState) {
             LocationTracker.getInstance().start(this);
         }
 
+        handleRecording(getIntent());
+    }
+
+    private void handleRecording(Intent intent) {
+        if (intent == null) {
+            Log.e(TAG, "handleRecording: Unable to retrieve intent instance");
+            finishActivity();
+            return;
+        }
+
+        String recordingAction = intent.getAction();
+        if (recordingAction == null) {
+            Log.e(TAG, "handleRecording: Unable to retrieve intent.action instance");
+            finishActivity();
+            return;
+        }
+
+        if (!recordingAction.startsWith(ACTION_RECORDING_BASE)) {
+            Log.i(TAG, "handleRecording: Received different intent with action: "
+                    + recordingAction);
+            finishActivity();
+            return;
+        }
+
+        if (RecorderUtil.isLowerThanQ()) {
+            Log.e(TAG, "handleRecording: Current Android OS Version is lower than Q");
+            finishActivity();
+            return;
+        }
+
+        if (!RecorderUtil.areRecordingPermissionsGranted(getApplicationContext())) {
+            Log.e(TAG, "handleRecording: Required permissions are not granted");
+            finishActivity();
+            return;
+        }
+
+        if (recordingAction.equals(ACTION_RECORDING_START)) {
```

Comment:
would `switch` be more readable here?

Reply:
there are only 3 case (ACTION_START, ACTION_STOP and Unknown action) so `if-elif `or `switch-case `wouldn't make so much difference i guess, and also with `switch-case` we need to indent code blocks extra 1-tab right compared to `if-elif`

```
if (recordingAction.equals(ACTION_START)) {
    blablabla..
}
```

versus 

```
switch (recordingAction) {
    case ACTION_START:
        blablabla
        break;
}
```

but again i can change it if you want/prefer 👍